### PR TITLE
MAKE-1044: grip check function improvements

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: errors_multi_test.go [hunk 1]
```diff
@@ -346,3 +346,24 @@ func (s *CatcherSuite) TestCheckWhenNoError() {
 	s.catcher.Check(fn)
 	s.Equal(s.catcher.Len(), 0)
 }
+
+func (s *CatcherSuite) TestChecExtendNoError() {
```
Comment (julianedwards) [inline errors_multi_test.go]: typo

File: errors_multi_test.go [hunk 2]
```diff
@@ -346,3 +346,24 @@ func (s *CatcherSuite) TestCheckWhenNoError() {
...
+	fn := func() error { return nil }
+	s.catcher.CheckExtend([]CheckFunction{fn, fn, fn})
+	s.Equal(s.catcher.Len(), 0)
+}
+
+func (s *CatcherSuite) TestChecExtendError() {
```
Comment (julianedwards) [inline errors_multi_test.go]: typo

File: errors_multi_test.go [hunk 3]
```diff
@@ -346,3 +346,24 @@ func (s *CatcherSuite) TestCheckWhenNoError() {
...
+	fn := func() error { return errors.New("hi") }
+	s.catcher.CheckExtend([]CheckFunction{fn, fn, fn})
+	s.Equal(s.catcher.Len(), 3)
+}
+
+func (s *CatcherSuite) TestChecExtendMixed() {
```
Comment (julianedwards) [inline errors_multi_test.go]: typo

File: errors_multi.go [hunk 4]
```diff
@@ -52,8 +56,9 @@ type Catcher interface {
 	Wrap(error, string)
 	Wrapf(error, string, ...interface{})
 
-	Check(func() error)
-	CheckWhen(bool, func() error)
+	Check(CheckFunction)
```
Comment (julianedwards) [inline errors_multi.go]: why not combine `Check` and `CheckExtend` with something like `Check(...CheckFunction)`

Comment (tychoish) [reply errors_multi.go]: wanted to match the api of Add/AddExtend

**Pull request merged**